### PR TITLE
Move to rolling versions

### PR DESCRIPTION
**Pull request by lfdebrux**
This PR has two commits, that together form part of a line of work to make maintaining our base Docker images easier.

The first change is to move the python image we are based on to `python:3.6-slim-stretch`. Because this no longer pins the minor version, it means we don't have to change our Dockerfile regularly to keep up with the up-to-date images on Docker Hub.

The second change is to create an additional tag in our image build process, that includes the date of the build. This should make sure that we retain a modicum of history in our Docker images for tracing/reverting potential breakages.

Because the `VERSION` has also been changed in these commits, these changes won't affect apps until they upgrade their base image to `3.3.0`. If we don't want to go full rolling release straight away, we can pin apps to a specific version and build date; however, the vision is that at some point there will be a pipeline for upgrading Docker images for apps separate from changing their Dockerfiles. That's still some way off though.

### CODE REVIEW

**Comment by risicle** [comment]
Mmm _hmm_ :thinking:

**Comment by risicle** [comment]
I mean, it seems sensible to *me*

File: Makefile [hunk 1]
```diff
@@ -1,24 +1,31 @@
 BUILD_VERSION := $(shell cat VERSION)
+BUILD_DATE := $(shell date '+%Y%m%d')
```
Comment (Wynndow) [inline Makefile]: Could it be worth adding a timestamp to this? It's not unfeasible that we'd have two builds in one day?

**Comment by lfdebrux** [comment]
@risicle that is the highest compliment 😝

File: Makefile (hunk 1 above)
Comment (lfdebrux) [reply Makefile]: So the date format is based on what other images on Docker hub use.

I think that the idea is that we shouldn't regularly be building images more than once a day; I think a 3 day cadence would be more than enough, but that's a question for later.

If we need to build an image more than once in one day then I think that means we should also be updating the version number, because either we've had to rebuild it because something broke and it needed fixing, or we're making new releases because we've added new features.

*tl;dr* I don't think a timestamp adds much

Comment (risicle) [reply Makefile]: I think we might not *intend* to release more than once a day, but the problem is that screwups happen.

Comment (katstevens) [reply Makefile]: ➕for timestamp. Otherwise looks sensible to me 👍

**Comment by lfdebrux** [comment]
By popular demand, I've added a UTC time to the tag

**Comment by lfdebrux** [comment]
I got the thumbs up from @katstevens and @benvand offline, so I'm merging this now.

I'm also going to use the Makefile and confirm that the `latest` tag on Docker Hub is updated correctly, as currently it is 4 months behind the actual latest image.
